scripts/cangen/cangen/can_generator.py

```diff
@@ -247,13 +237,22 @@ def generate_code(bus: Bus, config: Config):
     logger.info("Code generation complete")
 
 
+def _prepare_output_directory(output_dir):
+    """Deletes previously generated files and creates a gitignore for the directory"""
+    os.makedirs(output_dir, exist_ok=True)
+    for file in os.listdir(output_dir):
+        os.unlink(file)
```

Comment:
`file` is relative to `output_dir` but `os.unlink` is relative to the current dir (ex. projects/EV5/TMS) so this removes the wrong file. Also unlink will not work if `generated/can` contains folders at some point

I think we should go back to something like what we had before:
```
if (output dir exists):
    rmtree(output_dir)

makedirs(output_dir)
create .gitignore
```

sorry for going backwards, i assumed there was a simple function for clearing a folder without deleting it but that doesn't seem to be the case.